### ISSUE TITLE
Package structure

### ISSUE THREAD

**Issue by magnusnissel** [buzz-word/tools#3]
I am a big fan of planning out the structure of a package ahead - while being flexible to adjust as needed, of course - and am generally in favor of a relatively "verbose" (sub-) package structure similar to sklearn (https://github.com/scikit-learn/scikit-learn/tree/master/sklearn).

What does everyone else think about this? If you agree, we can start planning out a prelim layout. Like most of you, I have so bits of code lying around that pending some minor refactoring could be added relatively quicky, e.g. a couple of keyword-related functions. Having the broad structure layed out would make it easy to add things like that early, then erase/rebuild/improve as needed.

**Comment by interrogator** [comment]
The main thing I didn't commit to when making this repo is a specific programming language. I know we can all code in Python,  but I thought it could also be wise to remain a little bit language agnostic for now, or at least to bear in mind that some other languages might get used. So, it'd be nice if even if we are obviously moving toward Python, bearing in mind that other things may pop up.

But yeah, the `scikit-learn` package is a nice example to follow. 

If you wanted to contribute some basic structure for where our code will go, in return I'm happy to sort out:

- `setup.py`
- `pip` support
- `travis ci`
- `gitter` chat room

Since I've done all these things a few times before...

**Comment by magnusnissel** [comment]
That would be appreciated, I've only used very basic setup.py configurations so far.

I think given us as the initiators, Python is an obvious choice, but I would not be opposed to focusing on Python while also building an R version (at whatever pace we manage) and seeing if there is enough support for both to coexist. I don't think any other languages will be relevant. I think any new (core) features should first be implemented in Python unless there is a very compelling reason to switch it up.
All the philosophy & meta discussions should stay as agnostic as possible, I agree.

If we decide to support R, we should definitely invite https://github.com/cwolk

**Comment by interrogator** [comment]
Fair enough. Happy to develop the repo structurally as standard Python, while leaving open the possibility of other languages popping up and potentially being distributed/incorporated too
